__tests__/integration/test_parse.py

```
@@ -0,0 +1,99 @@
+import pytest
+from task_script_utils.datetime_parser import DatetimeConfig, parse
+from task_script_utils.datetime_parser.parser_exceptions import DatetimeParserError
+
+four_digit_year_with_no_config_test_cases = [
+    # input_, year_first, day_first, expected
+    # Unambiguous Cases
+    ("2021-12-13T12:12:12 America/Chicago", None, None, "2021-12-13T12:12:12-06:00"),
+    ("2021-13-12T12:12:12 America/Chicago", None, None, "2021-12-13T12:12:12-06:00"),
+    ("27-12-2002T12:12:12 America/Chicago", None, None, "2002-12-27T12:12:12-06:00"),
+    ("27-12-2002T12:12:12Z", None, None, "2002-12-27T12:12:12+00:00"),
+    ("2021-13-12T12:12:12Z", None, None, "2021-12-13T12:12:12+00:00"),
+    ("12-12-2002T12:12:12 America/Chicago", None, None, "2002-12-12T12:12:12-06:00"),
+
+
+    # Ambiguous / Error Cases
+    ("32-32-2002T12:12:12 America/Chicago", None, None, None),
+    ("32-31-2002T12:12:12 America/Chicago", None, None, None),
+    ("2021-23-13T12:12:12 America/Chicago", None, None, None),
+]
+
+four_digit_year_with_day_first_test_cases = [
+    # input_, year_first, day_first, expected
+    # Unambiguous Cases
+    ("2021-23-12T12:12:12 America/Chicago", None, True, "2021-12-23T12:12:12-06:00"),
+    ("2021-12-23T12:12:12 America/Chicago", None, False, "2021-12-23T12:12:12-06:00"),
+    ("27-12-2002T12:12:12 America/Chicago", None, True, "2002-12-27T12:12:12-06:00"),
+    ("11-12-2002T12:12:12 America/Chicago", None, False, "2002-11-12T12:12:12-06:00"),
+    ("2021-12-23T12:12:12Z", None, False, "2021-12-23T12:12:12+00:00"),
+
+    # Note: For short dates with 4 digits year, DatetimeConfig.year_first is ignored
```

Is this the right behavior? Should we actually throw an error? I'm thinking error is better because the user explicitly set this config however the data shows the other way. It's def not the user's expectation.